backend/app/adapter/gqlapi/resolver/authmutation.go

```diff
@@ -77,16 +34,15 @@ func (a AuthMutation) CreateShortLink(args *CreateShortLinkArgs) (*ShortLink, er
 		return nil, ErrInvalidAuthToken{}
 	}
 
-	longLink := args.ShortLink.longLink()
 	customAlias := args.ShortLink.CustomAlias
-	u := entity.ShortLink{
-		LongLink: longLink,
-		ExpireAt: args.ShortLink.ExpireAt,
+	u := args.ShortLink.GetShortLink()
+	if u == nil {
+		return nil, ErrUnknown{}
 	}
```

Comment:
Let's capture the error returned from `CreateShortLink` instead. Ultimately, it is usecase's decision to determine what to do when input is empty.

Reply:
With this change, we now dereference the `ShortLink` object before it's passed into `CreateShortLink`. I throw the `ErrUnknown` error at this point so that we don't get into the problem of dereferencing a null pointer. If we opt for throwing the error from `CreateShortLink` instead, then we should remove the `IsEmpty` method from `ShortLinkInput` and always return a `ShortLink` object from `GetShortLink`.

Reply:
@Coteh We will have to modify the signature and implementation both `CreateShortLink` and `UpdateShortLink`.